one_fm/overrides/employee.py

```diff
@@ -122,6 +124,20 @@ def validate_leaves(self):
                 '{getdate()}' BETWEEN from_date AND to_date
             """, as_dict=1):
             frappe.throw(f"Status cannot be 'Vacation' when no Leave Application exists for {self.employee_name} today {getdate()}.")
+            
+            
+def validate_employee_status_access(self):
+    if self.status:
+        if self.is_new():
+            pass
+        else:
+            if self.status != self.get_doc_before_save().status:
+                if not check_employee_access(email=frappe.session.user):
+                    frappe.throw("You are not allowed to make changes to an employee's status.")
+        
+def check_employee_access(email: str) -> bool:
+    employee_setting = frappe.get_doc("ONEFM General Setting").get("employee_access")
```

Comment:
Instead of get the complete object of the settings you can use
```
employees = frappe.get_all(
  "Employee Access", 
  fields=["employee"], 
  filters={
       "parenttype": "ONEFM General Settings"
  },
  order_by="idx"
)
```

When you use frappe.get_doc, it fetches a lot of data for the document including all the fields (1 query) and child table rows (1 query for each child table).

Reply:
oh, ok thanks, i will make changes